Replace "base" with "debian" in API documentation

thaJeztah opened the pull request:
### Update:

Following the discussion below, I changed the PR and removed the "proposal" (for now), I only updated the documentation and replaced "base" with "ubuntu" in the examples.

<strike>
The current documentation uses the "base" image in various places. The "base" image is deprecated and it is no longer possible to "pull" this image. (see https://github.com/docker/docker/issues/9775)
</strike>
<strike>
To prevent such cases in the future and to differentiate between "actual" and "illustrational" examples in the documentation, an "example" repository/image should be created on the registry that is meant for this purpose.
</strike>
<strike>
The "example" repository/image should have a number of frequently-used "tags" (e.g. `example:foo`, `example:bar`) and contain some "dummy" commands, to allow, for example, `docker run example baz --bar` to _illustrate_ an example and still be "functional" (it doesn't produce errors). The dummy commands could even output something, for example:
</strike>

```
"This output is produced by the 'baz' command running in the container."
```

<strike>
This proposal contains the initial change to the documentation style guide, and updates the API documentation to use the `example` image (actual response, image-IDs are not yet updated).
</strike>

<strike>
This is (obviously) NOT ready to be merged, because the example image does not yet exist and should be decided on.
</strike>

thaJeztah committed:
Replace "base" with "ubuntu" in documentation

The API documentation uses the "base" image in various
places. The "base" image is deprecated and it is no longer
possible to download this image.

This changes the API documentation to use "ubuntu" in stead.

Signed-off-by: Sebastiaan van Stijn <github@gone.nl>

thaJeztah commented:
And apparently, replacing `base` in the docs was done in in the past in https://github.com/docker/docker/pull/1914 but those changes got lost somehow

SvenDowideit commented:
we would really like to have all examples be functional and representative of the normal work. I'd rather replace `base` with `ubuntu`, and to make sure that we have some testing regime that helps us identify breakages.

I do think we need to start working towards something like this though - perhaps even using the `dockerdocs` organisation.

@fredlf @jamtur01

jamtur01 commented:
What @SvenDowideit said - I'd prefer `ubuntu`.

thaJeztah commented:
I'm fine with Ubuntu (#1914 chose to alternate ubuntu, debian and fedora). I think an "example" image could still be useful in cases where it only is a "partial" example and more clearly communicate it's just an example, not that `ubuntu` is a requirement.

I'll update this PR (hope it doesn't need a rebase) and use `ububtu[:latest]`. 

> I do think we need to start working towards something like this though - perhaps even using the dockerdocs organisation.

You mean, an example image? And is a "dockerdocs"

thaJeztah commented:
..(Hmpf, mobile) .. is a "dockerdocs" organization something fictional or actually something being worked on?

thaJeztah commented:
@SvenDowideit @jamtur01 PTAL

I updated the PR and removed the "proposal" (for now), I only updated the documentation and replaced "base" with "ubuntu" in the examples.

SvenDowideit commented:
LGTM - @fredlf @jamtur01 

`dockerdocs` org, or something like it is being thought about as a possible solution to some of the complexity we're facing with publishing documentation for more than just the Docker engine.

jamtur01 commented:
LGTM

thaJeztah commented:
> dockerdocs org, or something like it is being thought about as a possible solution to some of the complexity we're facing with publishing documentation for more than just the Docker engine

Ah, thanks for explaining, didn't hear of it yet.
